protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -395,22 +456,65 @@ impl BridgeContractCounterparty for MovementClient {
 		recipient: RecipientAddress<Self::Address>,
 		amount: Amount,
 	) -> BridgeContractCounterpartyResult<()> {
-		//@TODO properly return an error instead of unwrapping
+		
+		let test_noop_payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"noop_for_testing",
+			Vec::new(),
+			Vec::new(),
+		);
+
+		let _ = utils::send_and_confirm_aptos_transaction(
+			&self.rest_client,
+			self.signer.as_ref(),
+			test_noop_payload,
+		)
+		.await
+		.map_err(|_| BridgeContractCounterpartyError::LockTransferAssetsError);
+
+		let test_u64_args = vec![
+			bcs::to_bytes(&initiator.0).unwrap(),
+			bcs::to_bytes(&bridge_transfer_id.0[..]).unwrap(),
+			bcs::to_bytes(&hash_lock.0[..]).unwrap(),
+			bcs::to_bytes(&time_lock.0).unwrap(),
+			bcs::to_bytes(&recipient.0.0).unwrap(),
+			bcs::to_bytes(&amount.0).unwrap(),
+		];
+
+		let test_u64_payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"pass_data_for_testing",
+			Vec::new(),
+			test_u64_args,
+		);
+
+		let _ = utils::send_and_confirm_aptos_transaction(
+			&self.rest_client,
+			self.signer.as_ref(),
+			test_u64_payload,
+		)
+		.await
+		.map_err(|_| BridgeContractCounterpartyError::LockTransferAssetsError);
+
 		let args = vec![
-			to_bcs_bytes(&initiator.0).unwrap(),
-			to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
-			to_bcs_bytes(&hash_lock.0).unwrap(),
-			to_bcs_bytes(&time_lock.0).unwrap(),
-			to_bcs_bytes(&recipient.0).unwrap(),
-			to_bcs_bytes(&amount.0).unwrap(),
+			bcs::to_bytes(&initiator.0).unwrap(),
```

Comment:
Can we either `?` the result or `map_err(|_| BridgeContractCounterpartyError::SomeAppropriateVariant);`, we don't wont our production code to have unwraps as this would cause the service to crash.